metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/AggregationHelper.java

```
@@ -0,0 +1,125 @@
+        Optional<ExecutionStatusMetric> aggregatedExecutionStatus = getAggregatedExecutionStatus(executions);
+        if (getAggregatedExecutionStatus(executions).isPresent()) {
+            Metrics aggregatedMetrics = new Metrics();
+            aggregatedMetrics.setExecutionStatusCount(aggregatedExecutionStatus.get());
+            getAggregatedExecutionTime(executions).ifPresent(aggregatedMetrics::setExecutionTime);
+            getAggregatedCpu(executions).ifPresent(aggregatedMetrics::setCpu);
+            getAggregatedMemory(executions).ifPresent(aggregatedMetrics::setMemory);
+            return Optional.of(aggregatedMetrics);
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<ExecutionStatusMetric> getAggregatedExecutionStatus(List<Execution> executions) {
+        Map<String, Integer> statusCount = executions.stream()
+                .map(execution -> execution.getExecutionStatus().toString())
+                .collect(groupingBy(Function.identity(), Collectors.reducing(0, e -> 1, Integer::sum)));
+        // This shouldn't happen because all executions should have an execution status, but check anyway
+        if (statusCount.isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(new ExecutionStatusMetric().count(statusCount));
+    }
+
+    public static Optional<ExecutionTimeMetric> getAggregatedExecutionTime(List<Execution> executions) {
+        List<String> executionTimes = executions.stream()
+                .map(Execution::getExecutionTime)
+                .filter(Objects::nonNull)
+                .toList();
+        List<Double> executionTimesInSeconds = executionTimes.stream()
+                .map(executionTime -> {
+                    // Convert executionTime in ISO 8601 duration format to seconds
+                    try {
+                        return Long.valueOf(Duration.parse(executionTime).toSeconds()).doubleValue();
```

By the time we reach this point, have we already checked the format of `executionTime` elsewhere?

---

No, I consider the attempt of parsing it to a `Duration` as the checking

---

I moved `Duration.parse` to a separate function to make it more clear that the parsing is an act of checking the execution time format

---

If we can't parse an `executionTime`, is that value silently ignored, and the aggregation computed without it?  If so, it that what's intended, or should we somehow flag/log/notify of that type of condition?

---

Added `executionTime` validation in the webservice, and made a small change in this PR so that we don't aggregate execution time metrics if there's malformed data (which is very unlikely to happen). Re-requesting your review